### ISSUE TITLE
Add country flag to Server Address

### ISSUE THREAD

**Issue by lostthing** [v2rayA/v2rayA#339]
### Is the feature related with existing problem?

No

### Content of the feature

Add server's country flag to Server Address column in web interface, this monkey script works https://greasyfork.org/en/scripts/437588-v2raya-add-country-flag-to-ip-or-domain/code , but it would be nice if v2rayA implements this function with cache or use better ip/domain -> country method.

**Comment by mzz2017** [comment]
The script is very nice! But I prefer to import this function as a plugin, just like your script. I'd like to add it to the documentation. How do you think?

**Comment by mzz2017** [comment]
But I found it did not work on my computer.
What does `@require     https://cdn.bootcss.com/jquery/3.4.1/jquery.min.js` do? I did not see the downloading of jquery in my network activity.
I am using `firefox 95.0.1`.
Thanks.

**Comment by lostthing** [comment]
Some jquery function was used in script. The [required](https://violentmonkey.github.io/api/metadata-block/#require) script will be downloaded along with installation and execute before the script.

I guess you are using Tampermonkey in Firefox, there is a bug of it.

Please try [Violentmonkey](https://addons.mozilla.org/en-US/firefox/addon/violentmonkey/), read more detailed [info](https://greasyfork.org/en/scripts/437588-v2raya-add-country-flag-to-ip-or-domain) of this script and update it.

**Comment by mzz2017** [comment]
It is cool and works fine for me.

**Comment by lostthing** [comment]
> The script is very nice! But I prefer to import this function as a plugin, just like your script. I'd like to add it to the documentation. How do you think?

Yes, it's ok to use the script as a plugin for v2rayA. It should be disabled as default since will send server info to 3rd party service. 

I think better way is get domain's ip locally first (ping/nslookup/..); 
Then convert ip to country(code), many opensource projects can do it offline; 
At last mapping country code to country flags, project like https://github.com/iMuFeng/round-flag-icons got very small size flags with svg format, so we can mapping offline too.

**Comment by mzz2017** [comment]
Yes, geoip.dat is exact an IP database.

**Comment by MarksonHon** [comment]
Closed: no activity for 60 days. Reopen or open a new issue if it still happens on the current release.
